code-reviews/recipes/Go.md

```diff
@@ -2,47 +2,55 @@
 
 ## Go Style Guide
 
-```gofmt``` is the automated code format style guide for Go.
-
-[Effective Go](https://golang.org/doc/effective_go.html) describes other standard elements of Go style we follow.
+[CSE](../../CSE.md) developers follow the [Effective Go](https://golang.org/doc/effective_go.html) Style Guide.
 
 ## Linters
 
-### ```gofmt```
+### go vet
+
+`go vet` is a static analysis tool that checks for common go errors, such as incorrect use of range loop variables or misaligned printf arguments. [CSE](../../CSE.md) Go code should be able to build with no `go vet` errors.
+
+### golint
+
+[golint](https://github.com/golang/lint) can be an effetive tool for finding many issues, but it errors on the side of false positives. It is best used by developers when working on code, not as part of an automated build process.
+
+## Automatic Code Formatting
```

Comment:
Is this not the same as linter?

Reply:
Do you mean golint, or the Automatic Code Formatting?  If you are referring to automatic code formatting, then no.  Code formatting will format the style of the code, maybe adding spaces, removing unused imports etc.  Linting usually looks for if the style is per specification, but often looks for other things as well, like logic errors, missing doc strings etc. like a compiler for interpreted languages.

Reply:
I was looking at gofmt that is marked as both linter and automatic code formatting tool. Code formatting as you describe it Is still linting to me. Linters will format the style of the code and fix spaces and remove unused imports (like tslint) . Is the code formatting separate from linters or can it also be part of the linter?

Reply:
ahh, I think in that case tslint is doing both linting and code formatting. most likely golint does too, I am not a go expert so can't say

Reply:
It was there before your change and I wasn't sure if you intentionally moved it to code formatting. It makes sense for me and it really depends on the language and tools we are using we can have both in the linter or have them separate. I never really thought about code formatting as separate probably because I never had to use a separate tool for it.